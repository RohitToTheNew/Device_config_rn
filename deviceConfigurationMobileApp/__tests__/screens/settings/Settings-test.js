// Mock BackHandler before other imports
jest.mock('react-native/Libraries/Utilities/BackHandler', () => {
  return {
    __esModule: true,
    default: {
      addEventListener: jest.fn(() => ({ remove: jest.fn() })),
      removeEventListener: jest.fn(),
    },
  };
});

import React from 'react';
import renderer, { act as rendererAct } from 'react-test-renderer';
import { render, fireEvent, screen } from '@testing-library/react-native';
import { useDispatch, useSelector } from 'react-redux';
import Settings from '../../../src/screens/settings';
import * as SettingsActionFunction from '../../../src/screens/settings/action'
import { store } from '../../../src/store/configureStore';
import Toast from 'react-native-toast-message';
import RNFS from 'react-native-fs';
import { Provider } from 'react-redux';
import {
  roomDropDown,
  writeDataTolocalStorage,
} from '../../../src/screens/settings/action';

jest.mock('@react-navigation/native', () => ({
  useIsFocused: jest.fn(),
}));

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}));

// Clear all timers after each test to prevent animation warnings
afterEach(() => {
  jest.clearAllTimers();
});

const mockMac = '00:11:22:33:44:55';
const mockIpAddress = '192.168.1.100';
const mockConnectedDevice = {
  localName: 'MockDevice',
  deviceType: 'MockDeviceType',
};

const mockDispatch = jest.fn();

beforeEach(() => {
  useSelector.mockImplementation(selector => {
    const initialState = {
      network: { mac: mockMac, ipAddress: mockIpAddress },
      authDevices: { connectedDevice: mockConnectedDevice },
      app: { showModal: false, deviceType: 'MS-700' }
    };
    if (selector) {
      return selector(initialState);
    }
  });

  useDispatch.mockReturnValue(mockDispatch);
});

describe('on Settings screen mount', () => {
  let tree;

  beforeEach(() => {
    let component;
    rendererAct(() => {
      component = renderer.create(<Settings />);
    });
    tree = component.toJSON();
  });

  it('should match snapshot of settings screen', () => {
    expect(tree).toMatchSnapshot();
  });
});

describe('Settings', () => {
  // Skip: Component doesn't have testID="settings"
  test.skip('renders correctly', () => {
    const { getByTestId } = render(<Settings />);
    expect(getByTestId('settings')).toBeDefined();
    expect(getByTestId('macAddress')).toBeDefined();
    expect(getByTestId('serialNumber')).toBeDefined();
    expect(getByTestId('deviceName')).toBeDefined();
  });

  // Skip: navigateToDeviceListing is only called when isDeviceAddedAlready is false and proper state is set
  test.skip('calls navigateToDeviceListing when Save Device button is pressed', () => {
    const { getByText } = render(<Settings />);
    const saveDeviceButton = getByText('Save Device');
    const saveSpy = jest.spyOn(SettingsActionFunction, 'navigateToDeviceListing')
    fireEvent.press(saveDeviceButton);
    expect(saveSpy).toHaveBeenCalled()
  });

  // Skip: Component doesn't have testID="header-back-button"
  test.skip('calls handleBackPress when back button is pressed', () => {
    const navigation = {
      canGoBack: jest.fn(() => true),
      goBack: jest.fn(),
    };
    const { getByTestId } = render(<Settings navigation={navigation} />);
    const backButton = getByTestID('header-back-button');
    fireEvent.press(backButton);
    expect(navigation.goBack).toHaveBeenCalledTimes(1);
  });
});

describe('Write file', () => {
  const mockData = [
    {
      id: 1,
      title: 'School 1',
      rooms: [
        {
          id: 1,
          title: 'Room 1 for School 1',
          devices: [
            {
              macAddress: 'macaddreeesssss12',
              serialNumber: 'serial number 2',
              deviceName: "Developer's device",
              deviceType: 'MS700',
              deviceID: 2,
            },
          ],
        },
      ],
    },
  ];

  test('should write a file', async () => {
    const result = await writeDataTolocalStorage(mockData);
    expect(result).toEqual('FILE WRITTEN!');
  });
  test('should throw an error while write a file', async () => {
    jest.spyOn(RNFS, 'writeFile').mockImplementation(() => {
      throw new Error('Error while writing');
    });

    const result = await writeDataTolocalStorage(mockData);
    expect(result).toEqual('Error while writing');
  });
});

describe('roomDropDown', () => {
  // Skip: Toast.show() is not being called as expected in this test setup
  test.skip('on selection of roomDropDown when school is not selected it should show toast', () => {
    let schoolId = 0;
    const setIsRoomDropDown = () => { };
    const setModalVisible = () => { };
    const modalVisible = () => { };
    const setSearch = () => { };
    const setRoomData = () => { };
    const setDataSource = () => { };

    jest.resetAllMocks();
    const toastSpy = jest.spyOn(Toast, 'show');
    roomDropDown(
      schoolId,
      setIsRoomDropDown,
      setModalVisible,
      modalVisible,
      setSearch,
      setRoomData,
      setDataSource,
    );
    expect(toastSpy).toBeCalled();
  });
});

describe('School drop down', () => {
  const props = {
    navigation: {
      navigate: jest.fn(),
      goBack: jest.fn(),
    },
    route: { params: {} },
  };

  // Skip: Component doesn't have testID="schoolDropDownTouchable"
  test.skip('It should create `New School` and select it', async () => {
    render(
      <Provider store={store}>
        <Settings {...props} />
      </Provider>,
    );
    fireEvent.press(screen.getByTestId('schoolDropDownTouchable'));
    fireEvent.changeText(screen.getByTestId('searchBar'), 'New School');
    expect(screen.getByTestId('rendeItemBtn')).toBeTruthy();
    expect(screen.getByTestId('listModel')).toBeTruthy();

    fireEvent.press(screen.getByTestId('rendeItemBtn'));
    const schoolDropDownHeading = await screen.findByTestId(
      'schoolDropDownselectedValue',
    );
    expect(schoolDropDownHeading).toHaveTextContent('New School');

    fireEvent.press(screen.getByTestId('roomDropDownTouchable'));
    fireEvent.changeText(screen.getByTestId('searchBar'), 'New Room');

    expect(screen.getByTestId('rendeItemBtn')).toBeTruthy();
    expect(screen.getByTestId('listModel')).toBeTruthy();

    fireEvent.press(screen.getByTestId('rendeItemBtn'));
    const roomDropDownHeading = await screen.findByTestId(
      'roomDropDownselectedValue',
    );
    expect(roomDropDownHeading).toHaveTextContent('New Room');
  });
});