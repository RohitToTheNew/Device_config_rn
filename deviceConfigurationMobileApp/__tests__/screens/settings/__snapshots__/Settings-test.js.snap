// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`on Settings screen mount should match snapshot of settings screen 1`] = `
<View
  style={
    {
      "backgroundColor": "#003E7E0D",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "marginHorizontal": 32,
        "marginTop": 46,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <
        height={48}
        style={
          {
            "marginEnd": 16,
          }
        }
        width={48}
      />
    </View>
    <Text
      numberOfLines={1}
      style={
        {
          "color": "#FB8C00",
          "flex": 1,
          "fontFamily": "CenturyGothic",
          "fontSize": 34,
          "fontWeight": "700",
        }
      }
    >
      [missing "en.settings" translation]
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <
        height={48}
        width={48}
      />
    </View>
  </View>
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "column",
        "paddingHorizontal": 30,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#FFFFFF0D",
          "borderBottomColor": "#808284",
          "borderBottomWidth": 2,
          "height": 120,
          "marginTop": 40,
          "width": "98%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "bottom": 2,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "position": "absolute",
          }
        }
      >
        <View
          style={
            {
              "borderBottomColor": "#808284",
              "borderBottomWidth": 0,
              "flex": 1,
            }
          }
        >
          <Text
            style={
              {
                "color": "#808284",
                "flex": 1,
                "fontFamily": "CenturyGothic",
                "fontSize": 30,
                "paddingBottom": 0,
                "paddingLeft": 0,
              }
            }
          >
            [missing "en.selectSchool" translation]
          </Text>
        </View>
        <View>
          <
            height={24}
            width={24}
          />
        </View>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#FFFFFF0D",
          "borderBottomColor": "#808284",
          "borderBottomWidth": 2,
          "height": 120,
          "marginTop": 40,
          "width": "98%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "bottom": 2,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "position": "absolute",
          }
        }
      >
        <View
          style={
            {
              "borderBottomColor": "#808284",
              "borderBottomWidth": 0,
              "flex": 1,
            }
          }
        >
          <Text
            style={
              {
                "color": "#B0B99990",
                "flex": 1,
                "fontFamily": "CenturyGothic",
                "fontSize": 30,
                "paddingBottom": 0,
                "paddingLeft": 0,
              }
            }
          >
            [missing "en.selectRoom" translation]
          </Text>
        </View>
        <View>
          <
            height={24}
            width={24}
          />
        </View>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#FFFFFF0D",
          "borderBottomColor": "#808284",
          "borderBottomWidth": 2,
          "height": 120,
          "marginTop": 40,
          "width": "98%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "bottom": 2,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "position": "absolute",
          }
        }
      >
        <View
          style={
            {
              "borderBottomColor": "#808284",
              "borderBottomWidth": 0,
              "flex": 1,
            }
          }
        >
          <Text
            style={
              {
                "color": "#808284",
                "flex": 1,
                "fontFamily": "CenturyGothic",
                "fontSize": 30,
                "paddingBottom": 0,
                "paddingLeft": 0,
              }
            }
          >
            MockDeviceType
          </Text>
        </View>
        <View>
          <
            height={24}
            width={24}
          />
        </View>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#FFFFFF0D",
          "height": 120,
          "marginTop": 40,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "bottom": 2,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "position": "absolute",
          }
        }
      >
        <View
          style={
            {
              "height": 112,
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "marginTop": 15,
              }
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "flex-end",
                    "borderBottomWidth": 1,
                    "borderColor": "#808284",
                    "borderRadius": 6,
                    "borderStyle": "solid",
                    "flexDirection": "row",
                    "height": 80,
                    "justifyContent": "space-between",
                    "paddingBottom": 12,
                    "width": 680,
                  },
                  undefined,
                ]
              }
            >
              <TextInput
                autoFocus={false}
                editable={true}
                enablesReturnKeyAutomatically={true}
                fieldName="Device Name"
                keyboardType="default"
                label="Device Name"
                labelStyle={
                  {
                    "backgroundColor": "transparent",
                  }
                }
                maxLength={45}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                style={
                  [
                    {
                      "color": "#1A1C1C",
                      "flex": 1,
                      "fontFamily": "CenturyGothic",
                      "fontSize": 30,
                      "fontWeight": "400",
                      "marginTop": 30,
                      "padding": 0,
                      "paddingStart": 1,
                      "textDecorationLine": "none",
                    },
                    undefined,
                  ]
                }
                value="MockDevice"
              />
            </View>
            <View
              collapsable={false}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                  "position": "absolute",
                  "top": -12,
                  "zIndex": -1,
                }
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  adjustsFontSizeToFit={true}
                  collapsable={false}
                  numberOfLines={1}
                  style={
                    {
                      "color": "#808284",
                      "fontFamily": "CenturyGothic",
                      "fontSize": 24,
                      "textAlign": "left",
                    }
                  }
                >
                  Device Name
                </Text>
              </View>
            </View>
            <Text
              style={
                [
                  {
                    "color": "#FF0000",
                    "fontFamily": "CenturyGothic-Bold",
                    "fontSize": 22,
                    "marginTop": 16,
                    "textAlign": "right",
                  },
                  undefined,
                  {
                    "textAlign": "right",
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#FFFFFF0D",
          "height": 120,
          "marginTop": 40,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "bottom": 2,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "position": "absolute",
          }
        }
      >
        <View
          style={
            {
              "borderBottomColor": "#808284",
              "borderBottomWidth": 1,
              "flex": 1,
            }
          }
        >
          <Text
            style={
              {
                "color": "#808284",
                "flex": 1,
                "fontFamily": "CenturyGothic",
                "fontSize": 22,
                "paddingLeft": 0,
                "paddingTop": 22,
              }
            }
          >
            Mac Address
          </Text>
          <Text
            style={
              {
                "color": "#1A1C1C",
                "flex": 1,
                "fontFamily": "CenturyGothic",
                "fontSize": 30,
                "paddingBottom": 0,
                "paddingLeft": 0,
              }
            }
          >
            00:11:22:33:44:55
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#FFFFFF0D",
          "height": 120,
          "marginTop": 40,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "bottom": 2,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "position": "absolute",
          }
        }
      >
        <View
          style={
            {
              "height": 112,
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "marginTop": 10,
              }
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "flex-end",
                    "borderBottomWidth": 1,
                    "borderColor": "#808284",
                    "borderRadius": 6,
                    "borderStyle": "solid",
                    "flexDirection": "row",
                    "height": 80,
                    "justifyContent": "space-between",
                    "paddingBottom": 12,
                    "width": 680,
                  },
                  undefined,
                ]
              }
            >
              <TextInput
                autoFocus={false}
                editable={true}
                enablesReturnKeyAutomatically={true}
                fieldName="Serial Number"
                keyboardType="default"
                label="Serial Number"
                labelStyle={
                  {
                    "backgroundColor": "transparent",
                  }
                }
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                style={
                  [
                    {
                      "color": "#1A1C1C",
                      "flex": 1,
                      "fontFamily": "CenturyGothic",
                      "fontSize": 30,
                      "fontWeight": "400",
                      "marginTop": 30,
                      "padding": 0,
                      "paddingStart": 1,
                      "textDecorationLine": "none",
                    },
                    undefined,
                  ]
                }
                value=""
              />
            </View>
            <View
              collapsable={false}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                  "position": "absolute",
                  "top": 28,
                  "zIndex": -1,
                }
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  adjustsFontSizeToFit={true}
                  collapsable={false}
                  numberOfLines={1}
                  style={
                    {
                      "color": "#808284",
                      "fontFamily": "CenturyGothic",
                      "fontSize": 30,
                      "textAlign": "left",
                    }
                  }
                >
                  Serial Number
                </Text>
              </View>
            </View>
            <Text
              style={
                [
                  {
                    "color": "#FF0000",
                    "fontFamily": "CenturyGothic-Bold",
                    "fontSize": 22,
                    "marginTop": 16,
                    "textAlign": "right",
                  },
                  undefined,
                  {
                    "textAlign": "right",
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "marginTop": 100,
          "padding": 40,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": true,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#B0B6BB33",
            "height": 100,
            "justifyContent": "center",
            "opacity": 1,
            "width": "50%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontFamily": "CenturyGothic",
              "fontSize": 30,
              "fontWeight": "700",
            }
          }
        >
          Save Device
        </Text>
      </View>
    </View>
  </View>
  <Modal
    backdrop={true}
    hideModal={[Function]}
    transparent={true}
    visible={true}
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#1A1C1CCC",
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 16,
            "maxWidth": 594,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            {
              "paddingBottom": 48,
              "paddingHorizontal": 55,
              "paddingTop": 64,
            }
          }
        >
          <Image
            source={
              {
                "default": "",
              }
            }
            style={
              {
                "alignSelf": "center",
                "height": 100,
                "marginBottom": 64,
                "width": 190,
              }
            }
          />
          <Text
            style={
              {
                "alignSelf": "center",
                "color": "#000",
                "fontFamily": "CenturyGothic",
                "fontSize": 30,
                "fontWeight": "700",
                "lineHeight": 38,
                "marginBottom": 32,
              }
            }
          >
            [missing "en.deviceDisconnected" translation]
          </Text>
          <Text
            style={
              [
                {
                  "color": "#484949",
                  "fontFamily": "CenturyGothic",
                  "fontSize": 26,
                  "fontWeight": "400",
                  "lineHeight": 30,
                  "textAlign": "center",
                },
                {
                  "color": "#484949",
                },
              ]
            }
          >
            [missing "en.turnOnYourBluetooth" translation]
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#003D7D",
                "flexDirection": "row",
                "height": 92,
                "justifyContent": "center",
                "marginTop": 32,
                "opacity": 1,
                "width": 484,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "alignItems": "center",
                    "color": "#FFFFFF",
                    "fontFamily": "CenturyGothic-Bold",
                    "fontSize": 34,
                    "fontWeight": "700",
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              [missing "en.turnOn" translation]
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignSelf": "center",
                "marginTop": 50,
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#003D7D",
                  "fontFamily": "CenturyGothic",
                  "fontSize": 26,
                  "fontWeight": "700",
                  "lineHeight": 30,
                  "textAlign": "center",
                }
              }
            >
              [missing "en.goToHome" translation]
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
  <Modal
    backdrop={true}
    hideModal={[Function]}
    transparent={true}
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#1A1C1CCC",
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 16,
            "maxWidth": 594,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            {
              "paddingBottom": 48,
              "paddingHorizontal": 55,
              "paddingTop": 64,
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#484949",
                  "fontFamily": "CenturyGothic",
                  "fontSize": 26,
                  "fontWeight": "400",
                  "lineHeight": 30,
                  "textAlign": "center",
                },
                {
                  "color": "#484949",
                },
              ]
            }
          >
            [missing "en.thisWillReboot" translation]
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#003D7D",
                "flexDirection": "row",
                "height": 92,
                "justifyContent": "center",
                "marginTop": 32,
                "opacity": 1,
                "width": 484,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "alignItems": "center",
                    "color": "#FFFFFF",
                    "fontFamily": "CenturyGothic-Bold",
                    "fontSize": 34,
                    "fontWeight": "700",
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              [missing "en.ok" translation]
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignSelf": "center",
                "marginTop": 50,
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#003D7D",
                  "fontFamily": "CenturyGothic",
                  "fontSize": 26,
                  "fontWeight": "700",
                  "lineHeight": 30,
                  "textAlign": "center",
                }
              }
            >
              [missing "en.cancel" translation]
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
